MusicCrawler.Lib/Services/Singletons/RecommendationInteractor.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace MusicCrawler.Lib.Services.Singletons;
+
+/**
+ * requirements:
+ *   https://github.com/Noggog/MusicCrawler/issues/16
+ * TODO: Should depend on some derivative of user input, for when the user wants to stop getting recommendations for certain artists.
+ */
+public class RecommendationInteractor
+{
+    private readonly IRecommendationRepo _recommendationRepo;
+    private readonly ILibraryQuery _libraryQuery;
+
+    public RecommendationInteractor(
+        IRecommendationRepo recommendationRepo,
+        ILibraryQuery libraryQuery)
+    {
+        _recommendationRepo = recommendationRepo;
+        _libraryQuery = libraryQuery;
+    }
+
+    public async Task<IEnumerable<Recommendation>> Recommendations()
+    {
+        var currentPlexLibrary = await _libraryQuery.QueryAllData();
+        var sourceArtists = currentPlexLibrary.TakeRandomly(10).Select(x => x.Metadata.Key).ToList();
```

Comment:
We'll want to evolve to either process the whole list, or take in the "page" of recommendations to process, if we decide to do a batched server processing pattern where each run processes a small amount.

But this should be fine for now